src/healthkit.cs

```diff
@@ -2442,4 +2566,225 @@ interface HKWorkoutRouteQuery {
 		IntPtr Constructor (HKWorkoutRoute workoutRoute, HKWorkoutRouteBuilderDataHandler dataHandler);
 	}
 
+	delegate void HKWorkoutBuilderCompletionHandler (bool success, NSError error);
+	[Watch (5,0), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface HKWorkoutBuilder
+	{
+		[NullAllowed, Export ("device", ArgumentSemantic.Copy)]
+		HKDevice Device { get; }
+
+		[NullAllowed, Export ("startDate", ArgumentSemantic.Copy)]
+		NSDate StartDate { get; }
+
+		[NullAllowed, Export ("endDate", ArgumentSemantic.Copy)]
+		NSDate EndDate { get; }
+
+		[Export ("workoutConfiguration", ArgumentSemantic.Copy)]
+		HKWorkoutConfiguration WorkoutConfiguration { get; }
+
+		[Protected]
+		[Export ("metadata", ArgumentSemantic.Copy)]
+		NSDictionary _Metadata { get; }
+
+		[Wrap ("_Metadata")]
+		HKMetadata Metadata { get; }
+
+		[Export ("workoutEvents", ArgumentSemantic.Copy)]
+		HKWorkoutEvent[] WorkoutEvents { get; }
+
+		[Export ("initWithHealthStore:configuration:device:")]
+		IntPtr Constructor (HKHealthStore healthStore, HKWorkoutConfiguration configuration, [NullAllowed] HKDevice device);
+
+		[Async]
+		[Export ("beginCollectionWithStartDate:completion:")]
+		void BeginCollection (NSDate startDate, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async]
+		[Export ("addSamples:completion:")]
+		void Add (HKSample[] samples, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async]
+		[Export ("addWorkoutEvents:completion:")]
+		void Add (HKWorkoutEvent[] workoutEvents, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async, Protected]
+		[Export ("addMetadata:completion:")]
+		void Add (NSDictionary metadata, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async]
+		[Wrap ("Add (metadata.Dictionary, completionHandler)")]
+		void Add (HKMetadata metadata, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async]
+		[Export ("endCollectionWithEndDate:completion:")]
+		void EndCollection (NSDate endDate, HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Async]
+		[Export ("finishWorkoutWithCompletion:")]
+		void FinishWorkout (HKWorkoutBuilderCompletionHandler completionHandler);
+
+		[Export ("discardWorkout")]
+		void DiscardWorkout ();
+
+		[Export ("elapsedTimeAtDate:")]
+		double GetElapsedTimeAtDate (NSDate date);
```

Comment:
The `AtDate` suffix is not needed: `double GetElapsedTime (NSDate atDate)` looks better.